### ISSUE TITLE
modify_Tfunc branch: overhaul of temp_func_form

### ISSUE THREAD

**Issue by mnlevy1981** [kristenkrumhardt/MARBL#4]
1. Make `temp_func_form_opt` a per-autotroph (and per-zooplankton) option
1. Define `temp_func_form_iopt_power` for power function scaling
1. Coccolithophores should use the power function scaling

End result is that we don't need

```
          if (lexplicit_calcifier(Tfunc_ind)) then
            Tfunc(Tfunc_ind,:) = 0.12_r8 * (max(c0, min(temperature(:), 27.0_r8))**0.4_r8)
          else
            Tfunc(Tfunc_ind,:) = Q_10**(((temperature(:) + T0_Kelvin) - (Tref + T0_Kelvin)) / c10)
          end if
```

in the Q10 block because that isn't a form of Q10 scaling.

**Comment by kristenkrumhardt** [comment]
Mike does step 1, then Kristen steps 2 and 3

**Comment by mnlevy1981** [comment]
In 6c1e0b6 I did step (1) and also laid the ground work for step (2) -- I defined a `temp_func_form_iopt_power` variable but didn't try to parameterize the function. `compute_temperature_functional_form()` contains the following:

```
    do Tfunc_ind = 1, size(Tfunc, dim=1)
      select case (temp_func_form_iopt(Tfunc_ind))
        case (temp_func_form_iopt_q10)
          Tfunc(Tfunc_ind,:) = Q_10**(((temperature(:) + T0_Kelvin) - (Tref(Tfunc_ind) + T0_Kelvin)) / c10)
        case (temp_func_form_iopt_arrhenius)
          Tfunc(Tfunc_ind,:) = exp(-Ea(Tfunc_ind) * (Tref(Tfunc_ind) - temperature(:)) &
                                   / (K_Boltz * (temperature(:) + T0_Kelvin) * (Tref(Tfunc_ind) + T0_Kelvin)))
        case (temp_func_form_iopt_power)
          Tfunc(Tfunc_ind,:) = 0.12_r8 * (max(c0, min(temperature(:), 27.0_r8))**0.4_r8)
      end select
    end do
```

Also, the current implementation doesn't let the default value of `Tref` depend on `temp_func_form_opt` when parsing the settings YAML file -- I'll need to make another commit with updated python logic to get the old behavior back.

**Comment by mnlevy1981** [comment]
> Also, the current implementation doesn't let the default value of `Tref` depend on `temp_func_form_opt` when parsing the settings YAML file -- I'll need to make another commit with updated python logic to get the old behavior back.

Keith recommended taking `Tref` out of the YAML file altogether; I'll keep it in the `settings_type` datatypes, but as a "derived" value dependent on the functional form -- so it will be 30 for `q_10`, 25 for `arrhenius`, and something else for `power`. Users will be required to use `SourceMods` to change it, which is consistent with its implementation prior to bringing in the SPECTRA model

**Comment by mnlevy1981** [comment]
Closed via #6
